### PR TITLE
feat(cohere): Cohere custom client support

### PR DESCRIPTION
**Pull request by debkanchan**
Adds support for directly using Cohere client through a `client` parameter enabling use of Cohere on Azure and Standalone Cohere instances

### CODE REVIEW

**Review by bracesproul — APPROVED** [review]
lgtm! Thanks for doing this

**Review by bracesproul — CHANGES_REQUESTED** [review]
looks like CI is failing. Could you push up fixes and re-request my review once done?

**Comment by debkanchan** [comment]
Sure, will do tomorrow

File: libs/langchain-cohere/src/rerank.ts [hunk 1]
```diff
@@ -25,6 +20,30 @@ export interface CohereRerankArgs {
...
+//   client: CohereClient;
+// };
+
+// export type CohereRerankArgs = CohereRerankApiKeyArgs | CohereRerankClientArgs;
+
+// function isClientArgs(
+//   fields: CohereRerankArgs
+// ): fields is CohereRerankClientArgs {
+//   return (fields as CohereRerankClientArgs).client !== undefined;
+// }
```
[the remark this answers is not in the document]
Comment (debkanchan) [reply libs/langchain-cohere/src/rerank.ts]: Done

**Review by bracesproul — APPROVED** [review]
lgtm as long as CI passes

**Comment by jacoblee93** [comment]
Looks great thanks!

**Pull request merged**